QuantConnect.ExanteBrokerage/ExanteBrokerage.cs

```diff
@@ -0,0 +1,845 @@
+        {
+            var symbolId = exanteFeedTrade.SymbolId;
+            if (!_subscribedTickers.TryGetValue(symbolId, out var symbol))
+            {
+                return null; // Not subscribed to this symbol.
+            }
+
+            if (exanteFeedTrade.Size == decimal.Zero)
+            {
+                return null;
+            }
+
+            // Convert the timestamp to exchange timezone and pass into algorithm
+            var time = GetRealTimeTickTime(exanteFeedTrade.Date, symbol);
+
+            var instrument = Client.GetSymbol(symbolId);
+
+            var size = exanteFeedTrade.Size ?? 0m;
+            var price = exanteFeedTrade.Price ?? 0m;
+            var tick = new Tick(time, symbol, "", instrument.Exchange, size, price);
+            return tick;
+        }
+
+        /// <summary>
+        /// Returns a timestamp for a tick converted to the exchange time zone
+        /// </summary>
+        private DateTime GetRealTimeTickTime(DateTime time, Symbol symbol)
+        {
+            DateTimeZone exchangeTimeZone;
+            if (!_symbolExchangeTimeZones.TryGetValue(symbol, out exchangeTimeZone))
+            {
+                // read the exchange time zone from market-hours-database
+                exchangeTimeZone = MarketHoursDatabase.FromDataFolder()
+                    .GetExchangeHours(symbol.ID.Market, symbol, symbol.SecurityType).TimeZone;
+                _symbolExchangeTimeZones.Add(symbol, exchangeTimeZone);
+            }
+
+            return time.ConvertFromUtc(exchangeTimeZone);
+        }
+
+        /// <summary>
+        /// Create a tick from the Exante tick shorts stream data
+        /// </summary>
+        /// <param name="exanteTickShort">Exante tick short stream data object</param>
+        /// <returns>LEAN Tick object</returns>
+        private Tick CreateTick(ExanteTickShort exanteTickShort)
+        {
+            if (!_subscribedTickers.TryGetValue(exanteTickShort.SymbolId, out var symbol))
+            {
+                // Not subscribed to this symbol.
+                return null;
+            }
+
+            var time = GetRealTimeTickTime(exanteTickShort.Date, symbol);
+            var bids = exanteTickShort.Bid.ToList();
+            var asks = exanteTickShort.Ask.ToList();
+            return new Tick(time, symbol, "", "",
+                bids.IsNullOrEmpty() ? decimal.Zero : bids[0].Size,
+                bids.IsNullOrEmpty() ? decimal.Zero : bids[0].Price,
+                asks.IsNullOrEmpty() ? decimal.Zero : asks[0].Size,
+                asks.IsNullOrEmpty() ? decimal.Zero : asks[0].Price);
+        }
+
+        #endregion
+
+        private void OnUserMessage(ExanteOrder exanteOrder)
+        {
+            Order order;
+            if (!_orderMap.TryGetValue(exanteOrder.OrderId, out order))
+            {
+                return;
+            }
+
+            Thread.Sleep(1_000); // Need to wait for `Client.GetTransactions(...)`
```

Comment:
Please, let me know if you know the better way to achieve the same result.

Reply:
Ah right, generally we shouldn't Thread.Sleep anywhere. Delaying an order event an entire second seems excessive I think, could we rely on the fee model instead? any deviation should be solved by the cash sync performed daily by Lean instead

Reply:
WDYT of an alternative of launching `while (true)` loop with `100ms` delay between calls until it gets the result?

Reply:
@Martin-Molinero Martin, ⬆️  ?

Reply:
@alexander-myltsev  We approximate some live trading for speed, fetching fees is one of those places as it can be easily modeled most of the time. @Martin-Molinero is saying to use the ExanteBrokerageFeeModel to avoid the call entirely. 

We are slowly systematically removing `thread.sleep()` from the LEAN engine and replacing it with ManualResetEvent's where required - -but this isn't a location which needs one. We just need to use the fee model.

Reply:
Fixed by 5b51d2a28c494285c0601a0e39959116956dcf5f. @Martin-Molinero Martin, please check